packages/fuel-gauge/src/blob-deploy.test.ts

```diff
@@ -369,26 +220,23 @@ describe('first try', () => {
 
     const receiver = Wallet.generate({ provider });
 
-    const factory = new ContractFactory(
-      PredicateFalseConfigurable.bytecode,
-      PredicateFalseConfigurable.abi,
-      wallet
-    );
-
-    const { waitForResult } = await factory.deployAsBlobTxForScript();
-    const { loaderBytecode, configurableOffsetDiff } = await waitForResult();
-    expect(loaderBytecode).to.not.equal(hexlify(PredicateFalseConfigurable.bytecode));
+    const loaderPredicate = await (
+      await new PredicateWithMoreConfigurables({ provider }).deploy(wallet)
+    ).waitForResult();
 
     const configurable = {
-      SECRET_NUMBER: 8000,
+      FEE: 99,
+      ADDRESS: getRandomB256(),
+      U16: 305,
+      U32: 101,
+      U64: 1000000,
+      BOOL: false,
     };
 
-    const newAbi = mapToLoaderAbi(PredicateFalseConfigurable.abi, configurableOffsetDiff);
-
     const predicate = new Predicate({
-      data: [configurable.SECRET_NUMBER],
-      bytecode: loaderBytecode,
-      abi: newAbi,
+      data: [configurable.FEE, configurable.ADDRESS],
+      bytecode: loaderPredicate.bytes,
+      abi: loaderPredicate.interface.jsonAbi,
```

Comment:
We should perhaps have a way of creating a new predicate from an existing one:
```ts
const loaderPredicate = await waitForDeploy();

const newLoaderPredicate = loaderPredicate.toNewInstance({ configurableConstants, data });
```
Because we currently don't have a nice way of duplicating a predicate but with new configurable constants and data.

Reply:
@nedsalk This makes complete sense. We need to allow users to use different Predicate configurable constants for the same deployed loader. Mind creating an issue for it?